src/nbl/video/CVulkanLogicalDevice.cpp

```diff
@@ -86,16 +121,37 @@ IDeviceMemoryAllocator::SMemoryOffset CVulkanLogicalDevice::allocate(const SAllo
 
     VkDeviceMemory vk_deviceMemory;
     bool isDedicated = (info.dedication != nullptr);
+    bool external = false;
     if(isDedicated)
     {
         // VK_KHR_dedicated_allocation is in core 1.1, no querying for support needed
         static_assert(MinimumVulkanApiVersion >= VK_MAKE_API_VERSION(0, 1, 1, 0));
         vk_allocateFlagsInfo.pNext = &vk_dedicatedInfo;
 
-        if(info.dedication->getObjectType() == IDeviceMemoryBacked::EOT_BUFFER)
+        if (info.dedication->getObjectType() == IDeviceMemoryBacked::EOT_BUFFER)
             vk_dedicatedInfo.buffer = static_cast<CVulkanBuffer*>(info.dedication)->getInternalObject();
-        else if(info.dedication->getObjectType() == IDeviceMemoryBacked::EOT_IMAGE)
+        else if (info.dedication->getObjectType() == IDeviceMemoryBacked::EOT_IMAGE)
             vk_dedicatedInfo.image = static_cast<CVulkanImage*>(info.dedication)->getInternalObject();
+
+        auto& ccParams = info.dedication->getCachedCreationParams();
+
+        if (ccParams.externalHandleType.value)
+        {
+            external = true;
+            // Importing
+            if (ccParams.externalHandle)
+            {
+                importInfo.handleType = VkExternalMemoryHandleTypeFlagBits(ccParams.externalHandleType.value);
+                importInfo.handle = ccParams.externalHandle;
+                vk_dedicatedInfo.pNext = &importInfo;
+            }
+            else // Exporting
+            {
+                exportInfo.handleTypes = VkExternalMemoryHandleTypeFlags(ccParams.externalHandleType.value);
+                vk_dedicatedInfo.pNext = &exportInfo;
+            }
+        }
```

Comment:
the limit about external memory needing to be dedicated is dictated by Vulkan/other APIs not because of how we've structured pre-existing Nabla code, right?

Reply:
Correct